src/dmd/arrayop.d

```diff
@@ -244,6 +244,22 @@ private void buildArrayOp(Scope* sc, Expression e, Objects* tiargs, Expressions*
     e.accept(v);
 }
 
+/***********************************************
+ * Some implicit casting can be performed by the _arrayOp template.
+ * Params:
+ *      tfrom = type converting from
+ *      tto   = type converting to
+ * Returns:
+ *      true if can be performed by _arrayOp
+ */
+bool isArrayOpImplicitCast(TypeDArray tfrom, TypeDArray tto)
+{
+    const tyf = tfrom.nextOf().toBasetype().ty;
+    const tyt = tto  .nextOf().toBasetype().ty;
+    return tyf == tyt ||
+           tyf == Tint32 && tyt == Tfloat64;
```

Comment:
Why the special casing ?

Reply:
It's the only one I know of. Can always add more if they come up.